TestWizardLocalVaultWarning test failure

## Description

The `TestWizardLocalVaultWarning` test in `cmd/tegata/tui_model_test.go` is failing. This test verifies that the wizard displays a local vault warning when a user attempts to create a vault on a non-removable drive (e.g., local `/tmp` directory).

## Requirements

- The test should detect non-removable paths correctly
- When a non-removable path is entered, the wizard should display a local vault advisory
- After the user confirms the advisory, the wizard should advance to the passphrase entry screen

## Expected outcome

**Current behavior:**
- Test expects `stateWizardWelcome` (state 0) after entering a non-removable path
- Actual behavior: receives state 5 instead
- This indicates the wizard is advancing beyond the welcome state when it should stop and show the warning

**Expected behavior:**
- Wizard should pause at `stateWizardWelcome` with `localVaultWarn=true`
- Second Enter should advance to `stateWizardPassphrase` and clear the warning flag

## Technical details

**Test location:** `cmd/tegata/tui_model_test.go:240-262`

**Failure output:**
```
--- FAIL: TestWizardLocalVaultWarning (0.00s)
    tui_model_test.go:243: expected stateWizardWelcome after first Enter on non-removable path, got 5
    tui_model_test.go:246: expected localVaultWarn=true after first Enter on non-removable path
    tui_model_test.go:251: expected stateWizardPassphrase after confirming local-drive advisory, got 5
```

**State machine:**
- State 0: `stateWizardWelcome`
- State 1: `stateWizardPassphrase`
- State 2: `stateWizardRecoveryKey`
- State 3: `stateWizardAddCredential`
- State 4: `stateWizardAuditOptIn`
- State 5: Unknown (likely `stateVaultList` or similar)

**Possible causes:**
1. The `isRemovablePath()` detection is not working correctly on the test environment (e.g., `/tmp` is being detected as removable)
2. The wizard state machine is bypassing the local vault warning logic
3. Test assumptions about path removability need adjustment for the current system

This is a pre-existing issue discovered during code review of PR #76.

--- a/cmd/tegata/tui_model_test.go
+++ b/cmd/tegata/tui_model_test.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"path/filepath"
 	"strings"
 	"testing"
 	"time"
@@ -232,12 +233,15 @@ func TestWizardRecoveryKeyBlocksDuringCreation(t *testing.T) {
 // TestWizardLocalVaultWarning asserts that choosing a non-removable path shows
 // the local-drive advisory on the first Enter and advances on the second Enter.
 func TestWizardLocalVaultWarning(t *testing.T) {
+	tmpDir := t.TempDir()
+	vaultPath := filepath.Join(tmpDir, "test-vault")
+
 	m := initialModel("")
 	// Type a path that is definitely not on a removable drive.
-	m = typeInto(m, "/tmp/test-vault")
+	m = typeInto(m, vaultPath)
 	// First Enter: advisory should be shown; state stays at welcome.
 	m = sendKey(m, "enter")
-	if !isRemovablePath("/tmp/test-vault") {
+	if !isRemovablePath(vaultPath) {
 		// Non-removable: expect the warning state.
 		if m.state != stateWizardWelcome {
 			t.Errorf("expected stateWizardWelcome after first Enter on non-removable path, got %v", m.state)
@@ -264,11 +268,14 @@ func TestWizardLocalVaultWarning(t *testing.T) {
 // TestWizardLocalVaultWarningClearsOnEdit asserts that editing the path after
 // the advisory clears the warning so a fresh check runs on the next Enter.
 func TestWizardLocalVaultWarningClearsOnEdit(t *testing.T) {
+	tmpDir := t.TempDir()
+	vaultPath := filepath.Join(tmpDir, "test-vault")
+
 	m := initialModel("")
-	m = typeInto(m, "/tmp/test-vault")
+	m = typeInto(m, vaultPath)
 	m = sendKey(m, "enter") // trigger advisory (if non-removable)
 	if !m.localVaultWarn {
-		t.Skip("skipping: /tmp is removable on this system")
+		t.Skip("skipping: path is removable on this system")
 	}
 	// Typing clears the advisory.
 	m = typeInto(m, "x")

